packages/account/src/Sections/Verification/ProofOfIdentity/onfido-sdk-view.tsx

```diff
@@ -1,22 +1,59 @@
-import * as React from 'react';
+import React from 'react';
+import classNames from 'classnames';
 import countries from 'i18n-iso-countries';
 import * as Cookies from 'js-cookie';
-import { init } from 'onfido-sdk-ui';
-import { Loading, Text, ThemedScrollbars } from '@deriv/components';
-import { isMobile, WS, routes } from '@deriv/shared';
+import { init, SdkHandle, SdkResponse, SupportedLanguages } from 'onfido-sdk-ui';
+import { CSSTransition } from 'react-transition-group';
+import { GetSettings, ResidenceList } from '@deriv/api-types';
+import { HintBox, Loading, Text, ThemedScrollbars } from '@deriv/components';
+import { isMobile, WS } from '@deriv/shared';
 import { getLanguage, Localize } from '@deriv/translations';
 import ErrorMessage from 'Components/error-component';
 import getOnfidoPhrases from 'Constants/onfido-phrases';
 import MissingPersonalDetails from 'Components/poi/missing-personal-details';
-import { Link } from 'react-router-dom';
+import PoiConfirmWithExampleFormContainer from 'Components/poi/poi-confirm-with-example-form-container';
 
-const OnfidoSdkView = ({ country_code, documents_supported, handleViewComplete, height, is_from_external }) => {
-    const [api_error, setAPIError] = React.useState();
-    const [onfido_service_token, setOnfidoToken] = React.useState();
-    const [missing_personal_details, setMissingPersonalDetails] = React.useState(false);
+type TAPI_error = {
+    code?: string;
+    message?: string;
+    type?: string;
+};
+
+type TService_token = {
+    error: TAPI_error;
+    service_token: { onfido: { token: string } };
+};
+
+type TOnfidoSdkView = {
+    account_settings: GetSettings;
+    country_code: string;
+    documents_supported:
+        | string[]
+        | DeepRequired<ResidenceList>[0]['identity']['services']['onfido']['documents_supported'];
+    getChangeableFields: () => string[];
+    handleViewComplete: () => void;
+    height?: number | string;
+    is_external?: boolean;
+};
```

Comment:
let's extract all the types to a different files, we need to aim for a clear code and smaller components which are easy to read